addons/zylann.scene_code_converter/converter.gd

```diff
@@ -158,24 +145,22 @@ static func _get_property_set_code(obj: Object, property_name: String, value) ->
 
 static func _value_to_code(v) -> String:
 	match(typeof(v)):
+		# var2str will automatically pad decimals for floats
+		# see https://github.com/godotengine/godot-proposals/issues/1693
 		TYPE_BOOL:
-			if v:
-				return "true"
-			else:
-				return "false"
+			return "true" if v else "false"
+		TYPE_REAL:
+			return str(var2str(v), "f")
 		TYPE_VECTOR2:
-			return str("Vector2(", v.x, ", ", v.y, ")")
+			return str("Vector2(", var2str(v.x), "f, ", var2str(v.y), "f)")
 		TYPE_VECTOR3:
-			return str("Vector3(", v.x, ", ", v.y, ", ", v.z, ")")
+			return str("Vector3(", var2str(v.x), "f, ", var2str(v.y), "f, ", var2str(v.z), "f)")
 		TYPE_COLOR:
-			return str("Color(", v.r, ", ", v.g, ", ", v.b, ", ", v.a, ")")
+			return str("Color(", var2str(v.r), "f, ", var2str(v.g), "f, ", var2str(v.b), "f, ", var2str(v.a), "f)")
 		TYPE_STRING:
 			return str("TTR(\"", v, "\")")
 		TYPE_OBJECT:
-			if v is Resource:
-				return "nullptr /* TODO resource here */"
-			else:
-				return "nullptr /* TODO reference here */"
+				return "/* TODO: reference here */"
```

Comment:
I removed the `nullptr` because it's easier to remember to implement something when the compiler is giving you an error about it